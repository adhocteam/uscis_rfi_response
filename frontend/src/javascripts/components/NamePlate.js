import React from "react";
import history from "../services/history";
import { isLoggedIn, getEmail } from "../helpers/session";

class NamePlate extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loggedIn: isLoggedIn(),
      email: getEmail(),
    };
    // on route change, update state
    history.listen(() => {
      this.setState({
        loggedIn: isLoggedIn(),
        email: getEmail(),
      });
    });
  }

  render() {
    const { loggedIn, email } = this.state;
    if (loggedIn) {
      return (
        <div className="ds-u-fill--primary-darker">
          <div className="ds-l-container">
            <div className="ds-l-row">
              <div className="ds-l-col--12">
                <p className="ds-u-margin-top--1 ds-u-margin-bottom--1 ds-u-color--white qa-uscis-name-plate">
                  Hello! You are logged in as{" "}
                  <span className="ds-u-font-weight--semibold">{email}</span>.
                </p>
              </div>
            </div>
          </div>
        </div>
      );
    } else {
      return null;
    }
  }
}

export default NamePlate;
