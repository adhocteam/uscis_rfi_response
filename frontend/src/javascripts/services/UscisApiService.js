import history from "./history";

const updateIfPresent = (headers, headerName, sessionName) => {
  const val = headers.get(headerName);
  if (val) {
    sessionStorage.setItem(sessionName, val);
  }
};

const updateTokens = headers => {
  updateIfPresent(headers, "access-token", "token");
  updateIfPresent(headers, "expiry", "expiry");
  updateIfPresent(headers, "uid", "uid");
  updateIfPresent(headers, "client", "client");
};

const authedRequest = (path, settings, error) =>
  fetch(`${BASE_URL}${path}`, {
    headers: {
      "Content-Type": "application/json",
      Accept: "application/json",
      "access-token": sessionStorage.getItem("token"),
      client: sessionStorage.getItem("client"),
      uid: sessionStorage.getItem("uid"),
      expiry: sessionStorage.getItem("expiry"),
    },
    ...settings,
  }).then(resp => {
    if (resp.ok) {
      updateTokens(resp.headers);
      return resp.json();
    }
    if (resp.status === 401) {
      history.replace(`/login?next=${history.location.pathname}`);
    }
    throw new Error(error);
  });

const UscisApiService = {
  getAdmin: () => authedRequest("/admin", {}, "Failed to get admin."),

  // TODO: error handling
  getSignedUrl: (user_id, image_name, image_type) => {
    return fetch(`${BASE_URL}/submissions/presigned_url`, {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        submission_id: submission_id,
        image_name: image_name,
        image_type: image_type,
      }),
    }).then(resp => {
      if (resp.status === 200) {
        return resp.json();
      }
    });
  },

  filterSubmissions: status => {
    return authedRequest(
      `/submissions/filter?status=${status}`,
      {},
      "Failed to get submissions."
    );
  },

  getSubmission: id =>
    authedRequest(`/submissions/${id}`, {}, `Failed to get submission ${id}.`),

  getSubmissions: () =>
    authedRequest("/submissions", {}, "Failed to get submissions."),

  login: (email, password) => {
    return fetch(`${BASE_URL}/auth/sign_in`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
      body: JSON.stringify({
        email: email,
        password: password,
      }),
    }).then(resp => {
      if (resp.ok) {
        updateTokens(resp.headers);
        return resp.json();
      } else {
        throw new Error("Failed to log in.");
      }
    });
  },

  updateSubmission: ({ id, notes, status }) => {
    authedRequest(
      `/submissions/${id}`,
      {
        method: "PUT",
        body: JSON.stringify({ notes, status }),
      },
      `Failed to update submission ${id}.`
    );
  },

  createUser: (name, dob, email, street1, street2, city, state, zip) => {
    return authedRequest(`submissions/new_upload`, {
      method: "POST",
      body: JSON.stringify({
        name: name || "",
        dob: dob || "",
        email: email || "",
        street1: street1 || "",
        street2: street2 || "",
        city: city || "",
        state: state || "",
        zip: zip || "",
      }),
    });
  },
};

export default UscisApiService;
