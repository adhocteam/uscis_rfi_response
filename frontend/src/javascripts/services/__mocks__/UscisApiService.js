const UscisApiService = {
  login: (email, password) => {
    sessionStorage.setItem("token", "FAKE-ACCESS-TOKEN");
    sessionStorage.setItem("client", "FAKE-CLIENT-ID");
    sessionStorage.setItem("uid", email);
    sessionStorage.setItem("expiry", Date.now() + 1209600000); //two weeks
    return Promise.resolve();
  },
};

export default UscisApiService;
