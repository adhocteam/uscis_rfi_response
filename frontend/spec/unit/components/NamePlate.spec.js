jest.mock("services/UscisApiService");

import React from "react";
import { shallow } from "enzyme";
import NamePlate from "components/NamePlate.js";
import UscisApiService from "services/UscisApiService.js";

test("renders nothing when logged out", () => {
  const wrapper = shallow(<NamePlate />);
  expect(wrapper.find(".qa-uscis-name-plate")).toHaveLength(0);
  expect(wrapper).toMatchSnapshot();
});

test("renders content when logged in", () => {
  UscisApiService.login("fake.admin@adhocteam.us", "thisisafakepassword");
  const wrapper = shallow(<NamePlate />);
  expect(wrapper.find(".qa-uscis-name-plate")).toHaveLength(1);
  expect(wrapper).toMatchSnapshot();
});
