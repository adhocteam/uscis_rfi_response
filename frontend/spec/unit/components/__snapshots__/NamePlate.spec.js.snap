// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content when logged in 1`] = `
<div
  className="ds-u-fill--primary-darker"
>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row"
    >
      <div
        className="ds-l-col--12"
      >
        <p
          className="ds-u-margin-top--1 ds-u-margin-bottom--1 ds-u-color--white qa-uscis-name-plate"
        >
          Hello! You are logged in as
           
          <span
            className="ds-u-font-weight--semibold"
          >
            fake.admin@adhocteam.us
          </span>
          .
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders nothing when logged out 1`] = `""`;
